### PR TITLE
ICMSLST-1833 Data Migration - Certificate Application Templates

### PR DESCRIPTION
**Pull request by mikeylockwood**

### CODE REVIEW

File: data_migration/queries/export_application/cat.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+com_template_timestamp_update = """
+UPDATE web_certificateofmanufactureapplicationtemplate
+SET last_update_datetime = dm_cat.last_updated_datetime
+FROM data_migration_certificateapplicationtemplate dm_cat
+WHERE web_certificateofmanufactureapplicationtemplate.template_id = dm_cat.id
+"""
+
+
+cfs_schedule_template_timesheet_update = """
```
Comment (MattHolmes123) [inline data_migration/queries/export_application/cat.py:90]: timesheet ?

File: web/domains/cat/models.py [hunk 2]
```diff
@@ -17,9 +17,9 @@
 
 class CertificateApplicationTemplate(models.Model):
     class SharingStatuses(TypedTextChoices):
-        PRIVATE = ("private", "Private (do not share)")
-        VIEW = ("view", "Share (view only)")
-        EDIT = ("edit", "Share (allow edit)")
+        PRIVATE = ("PRIVATE", "Private (do not share)")
+        VIEW = ("VIEW", "Share (view only)")
+        EDIT = ("EDIT", "Share (allow edit)")
```
Comment (MattHolmes123) [inline web/domains/cat/models.py:23]: These values are used in a few places that use the lowercase versions of these strings.
an end to end test
several templates 
a unittest.

Can we change them back to the lowercase version please so these places don't break

**Review by MattHolmes123 — DISMISSED** [review]
Added a few comments.

File: web/domains/cat/models.py (hunk 2 above)
Comment (MattHolmes123) [reply web/domains/cat/models.py:23]: In an ideal work all the places would use the enum values but in this instance they don't everywhere

Comment (MattHolmes123) [reply web/domains/cat/models.py:23]: I'm going to sumbit a PR to fix the app code, can you fix the broken end to end test as that code doesn't import any of the web code

Comment (MattHolmes123) [reply web/domains/cat/models.py:23]: https://github.com/uktrade/icms/pull/1465

File: data_migration/queries/export_application/cat.py (hunk 1 above)
Comment (mikeylockwood) [reply data_migration/queries/export_application/cat.py]: Timestamp. Was filling out my timesheet when I wrote this...

File: web/domains/cat/models.py (hunk 2 above)
Comment (mikeylockwood) [reply web/domains/cat/models.py:23]: Merged it in a squashed commits

File: data_migration/queries/export_application/cat.py (hunk 1 above)
Comment (MattHolmes123) [reply data_migration/queries/export_application/cat.py]: haha that's anazing :D

File: data_migration/management/commands/config/data_counts.py [hunk 3]
```diff
@@ -529,6 +529,21 @@
...
+    CheckQuery(
+        name="COM Application Templates",
+        query=queries.com_application_templates_count,
+        model=web.CertificateOfManufactureApplicationTemplate,
+    ),
+    CheckQuery(
+        name="GMP Application Templates",
+        query=queries.gmp_application_templates_count,
+        model=web.CertificateOfGoodManufacturingPracticeApplicationTemplate,
+    ),
```
Comment (mikeylockwood) [inline data_migration/management/commands/config/data_counts.py:546]: These checks are run after the migration in prod and check the number of app templates in V1 and V2 are the same for each application type. For GMP, with no data migration written for GMP app templates so if one gets added, this check should catch it and fail.

File: data_migration/models/export_application/cat.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    )
+
+
+class CFSTemplateLegislation(MigrationBase):
+    cfsschedule = models.ForeignKey(CFSScheduleTemplate, on_delete=models.CASCADE)
+    productlegislation = models.ForeignKey(ProductLegislation, on_delete=models.CASCADE)
+
+    @classmethod
+    def get_excludes(cls) -> list[str]:
+        return ["cfsschedule_id"]
```
Comment (marcuspp) [inline data_migration/models/export_application/cat.py:127]: This doesn't call super where as `CertificateOfManufactureApplicationTemplate` does. Is it required?

**Review by marcuspp — APPROVED** [review]

File: data_migration/models/export_application/cat.py (hunk 4 above)
Comment (mikeylockwood) [reply data_migration/models/export_application/cat.py:127]: It's not required for this one as there are only 2 fields on the model and I just want to exclude one of them. The `super()` call will  automatically exclude all the `_xml` fields which is useful on the models where they are used but not needed here

**Pull request merged**